### PR TITLE
fix: don't double-count file uploads in worker pool

### PR DESCRIPTION
**Pull request by vbrown608**
This was receiving from the semaphore twice (`sem <- 1`). Added a test and it's looking much better.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -16,6 +16,6 @@ require (
 	github.com/pkg/errors v0.9.1
 	github.com/rsc/goversion v1.2.0
 	github.com/sirupsen/logrus v1.6.0
-	github.com/stretchr/testify v1.6.1
+	github.com/stretchr/testify v1.7.0
```
Comment (vbrown608) [inline go.mod:19]: Upgraded to get `require.ErrorIs`

**Review by mraerino — DISMISSED** [review]
looks good. the test seems to be very tied to implementation details. would there a way to use more higher-level primitives in the test and reduce the setup boilerplate?

**Comment by vbrown608** [comment]
The higher level primitives we have for creating the client currently only support `https://api.netlify.com/api/v1` as the API endpoint. I did some consolidation of the authorization setup to provide only what's needed for these tests.

**Review by keiko713 — APPROVED** [review]
🔄

**Pull request merged**